### ISSUE TITLE
ASS subtitles sometimes fail to render when playing MKV with ordered chapters

### ISSUE THREAD

**Issue by MarcG2** [Nevcairiel/LAVFilters#244]
When playing a MKV with ordered chapters, and there are subtitles that are timed to occur right after a chapter transitions, those subtitles won't render.  However they will render when using the chapter skip function. The issue goes away when switching off the ordered chapters flag in the MKV.  I'm using Klite 14.6.0 to test.  I'm not sure if this bug had always been present, or if it was introduced at some point.  I can provide a sample file if that helps any.

**Comment by Nevcairiel** [comment]
Test samples would be helpful. Otherwise not sure what to even look for.

**Comment by MarcG2** [comment]
Here an example MKV with ordered chapters.  

https://drive.google.com/open?id=1Q_tWih_8FXhXcwcPFuukvfn4h7Nwpm3Z

At 5:10, the first line of dialog doesn't render.  At the very next chapter, 5:44, the text on the sign doesn't render.

I've also experienced render issues when playing content with complex subtitle arrangements.  For example, if there's supposed to text panning across the screen, the text may flicker or disappear early.  Would this be the fault of the Subtitle renderer?  In my case I'm using DirectVobSub .

**Comment by clsid2** [comment]
For flickering subs, this may help:
DirectVosbSub settings > Subpic queue > uncheck "Allow dropping some subpictures"

**Comment by Syst3mSh0ck** [comment]
I find madVR + xySubFilter better for 'complex subtitle arrangements'. DirectVobSub isn't actively maintained.

**Comment by MarcG2** [comment]
Thanks for the suggestions.  I enabled "allow dropping" setting, but it hasn't made a noticeable difference.
 I gave xySubFilter a try, but it has it has it's own unique issues.  I suspect its using a different implementation of certain font effects which cause it to render some text incorrectly.  So far, it seems like VLC handles rendering the most reliably.  But VLC has issues with MKVs with ordered chapters as well.
